source/sink/sink_impl.cc

```diff
@@ -104,4 +113,30 @@ FileSinkImpl::LoadExecutionResult(absl::string_view execution_id) const {
   return responses;
 }
 
+absl::Status
+InMemorySinkImpl::StoreExecutionResultPiece(const nighthawk::client::ExecutionResponse& response) {
+  absl::Status status = validateKey(response.execution_id(), false);
+  if (!status.ok()) {
+    return status;
+  }
+  auto iterator =
+      map_.insert({response.execution_id(), std::vector<nighthawk::client::ExecutionResponse>()})
+          .first;
+  iterator->second.push_back(response);
+  return absl::Status();
```

Comment:
I think the more conventional way of returning OK is:

```cc
return absl::OkStatus();
```

This one also works, but requires the reader to understand that an empty status defaults to OK.

https://abseil.io/blog/2020-091021-status